src/main/starlark/builtins_bzl/bazel/java/bazel_java_binary.bzl

```diff
@@ -41,7 +41,11 @@ def _bazel_base_binary_impl(ctx, is_test_rule_class):
     executable = _get_executable(ctx)
 
     feature_config = helper.get_feature_config(ctx)
-    strip_as_default = helper.should_strip_as_default(ctx, feature_config)
+    if feature_config:
+        strip_as_default = helper.should_strip_as_default(ctx, feature_config)
```

Comment:
It seems possible that this branch always evaluates to `False` in Bazel, in which case most of this logic could probably go away.